### DISCUSSION TITLE
Winbar support?

### DISCUSSION THREAD

**Discussion by lkhphuc** [nvim-lualine/lualine.nvim#688] [Ideas]
I'm surprised no one has asked this yet, but is there any plan to support the new [`winbar`](https://github.com/neovim/neovim/pull/18562)? 

Personally I would like to just open my lualine config, and move some sections to a `winbar = {}` table.

**Comment by diegodox** [comment]
I wrote an PR #689 today. You can try.
Add `winbar = {}` and `inactive_wibar = {}` to your config.

**Reply by shadmansaleh** [reply]
For now there isn't a plan to add winbar support maybe in future. See [here](https://github.com/nvim-lualine/lualine.nvim/pull/689#issuecomment-1133937406)

**Comment by rockyzhang24** [comment]
The most useful points of `winbar` are:
1. Full path of the filename (`%F`)
2. LSP symbol path (`aerial.nvim` has API to output this)
3. Window number for fast window switching (`vim.api.nvim_win_get_number()`)

Setting them is very easy and no need further supports. I believe a fancy `statusline` is necessary but a plus fancy `winbar` at the top will be distracting. Also, a verbose and complicated `winbar` will get truncated and making it meaningless at all.

<img width="1726" alt="Screen Shot 2022-05-23 at 15 53 56" src="https://user-images.githubusercontent.com/11582667/169917499-9f19886c-6a80-4ac0-8206-00792632a9da.png">

**Reply by milanglacier** [reply]
> The most useful points of `winbar` are:
> 
> 1. Full path of the filename (`%F`)
> 2. LSP symbol path (`aerial.nvim` has API to output this)
> 3. Window number for fast window switching (`vim.api.nvim_win_get_number()`)
> 
> Setting them is very easy and no need further supports. I believe a fancy `statusline` is necessary but a plus fancy `winbar` at the top will be distracting. Also, a verbose and complicated `winbar` will get truncated and making it meaningless at all.
> 
> <img alt="Screen Shot 2022-05-23 at 15 53 56" width="1726" src="https://user-images.githubusercontent.com/11582667/169917499-9f19886c-6a80-4ac0-8206-00792632a9da.png">

To implement a useful winbar like your screenshot  does not need a plugin at all, I just implement a simple winbar function by my own, and I directly link the highlight group of  `winbar` to `lualine`.

For the code outline part, there is a plugin `nvim-navic`  which provides a function that will returns a string indicating the outline.
And just use `vim.fn.expand` to get the current filename, combine the two strings together, and a winbar function is done!

**Comment by rockyzhang24** [comment]
@milanglacier 

> To implement a useful winbar like your screenshot does not need a plugin at all

That's what I meant, isn't it? For the symbol path, I use the API provided by aerial.nvim, because I am using it to display LSP symbols in a tree view (substituting symbol-outline.nvim), thus one plugin fits all. For the other sections, I only use the vanilla statusline items, i.e., no plugins.
